test/parallel/test-fs-read-stream-file-handle.js

```diff
@@ -80,3 +80,22 @@ fs.promises.open(file, 'r').then((handle) => {
     assert.strictEqual(output, input);
   }));
 }).then(common.mustCall());
+
+fs.promises.open(file, 'r').then((handle) => {
+  const controller = new AbortController();
+  const { signal } = controller;
+  const stream = handle.createReadStream({ signal });
+
+  stream.on('data', common.mustNotCall());
+  stream.on('end', common.mustNotCall());
+
+  stream.on('error', common.mustCall((err) => {
+    assert.strictEqual(err.name, 'AbortError');
+  }));
+
+  stream.on('close', common.mustCall(() => {
+    handle.close();
+  }));
+
+  controller.abort();
+}).then(common.mustCall());
```

Comment:
Nit: Should also include a variation on this test that uses an already-aborted signal to verify it is appropriately handled.

```js
fs.promises.open(file, 'r').then((handle) => {
  const signal = AbortSignal.abort();
  handle.createReadStream({ signal })
      .on('error', common.mustCall((err) => {
        assert.strictEqual(err.name, 'AbortError');
      }));
}).then(common.mustCall());
```